src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -17,16 +15,21 @@
 using System.Threading;
 using System.Collections.Concurrent;
 using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Hosting;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer;
+using Microsoft.PowerShell.EditorServices.Utility;
+using Microsoft.PowerShell.EditorServices.Services.Analysis;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
 
 namespace Microsoft.PowerShell.EditorServices.Services
 {
     /// <summary>
     /// Provides a high-level service for performing semantic analysis
     /// of PowerShell scripts.
     /// </summary>
-    public class AnalysisService : IDisposable
+    public class AnalysisService
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IDisposable' in this class and use the 'Dispose' method to call 'Dispose' on '_existingRequestCancellation'.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4416818)